services/registry/docker-compose.yml

```diff
@@ -66,7 +66,51 @@ services:
         mode: 0644
       - source: registry_config.yml
         target: /etc/docker/registry/config.yml
-
+  registry-pull-through-cache:
+    hostname: "registry-ptc-{{.Node.Hostname}}-{{.Task.Slot}}"
+    image: registry:2.8.2
+    environment:
+      REGISTRY_HTTP_SECRET: ${REGISTRY_PULL_THROUGH_CACHE_HTTP_SECRET}
+      # S3
+      REGISTRY_STORAGE: s3
+      REGISTRY_STORAGE_S3_ACCESSKEY: ${REGISTRY_S3_ACCESS_KEY}
+      REGISTRY_STORAGE_S3_SECRETKEY: ${REGISTRY_S3_SECRET_KEY}
+      REGISTRY_STORAGE_S3_REGIONENDPOINT: "${REGISTRY_S3_ENDPOINT}"
+      REGISTRY_STORAGE_S3_SECURE: "true"
+      REGISTRY_STORAGE_S3_BUCKET: ${REGISTRY_PULL_THROUGH_CACHE_S3_BUCKET}
+      REGISTRY_STORAGE_S3_REGION: ${REGISTRY_AWS_REGION}
```

Comment:
Question: what are the implications of this?
the pull-through registry also points to the private registry data correct?
since inside the deploys we only pull (never push), could it be that we could just use that one inside the deployments?
We only need one to push actually when we add images.

Reply:
pull-through registry does not point to private registry data

Reply:
I do not fully grasp the "one-side" way. Could you elaborate if you think it is still the case?